### PR TITLE
feat/per-block-door

### PR DESCRIPTION
**Pull request by ikbenignace**

### CODE REVIEW

File: Core/src/main/java/com/theprogrammingturkey/comz/game/actions/DoorSetupAction.java [hunk 1]
```diff
@@ -83,17 +83,35 @@ public void onPlayerInteractEvent(PlayerInteractEvent event)
 		if(door.arePointsFinal())
 			return;
 
-		if(event.getAction().equals(Action.LEFT_CLICK_AIR) || event.getAction().equals(Action.LEFT_CLICK_BLOCK))
+		// if(event.getAction().equals(Action.LEFT_CLICK_AIR) || event.getAction().equals(Action.LEFT_CLICK_BLOCK))
```
Comment (TheTurkeyDev) [inline Core/src/main/java/com/theprogrammingturkey/comz/game/actions/DoorSetupAction.java:86]: No need to comment this, just go ahead and delete it

File: Core/src/main/java/com/theprogrammingturkey/comz/game/features/Door.java [hunk 2]
```diff
@@ -185,6 +185,11 @@ public boolean arePointsFinal()
 		return arePointsFinal;
 	}
 
+	public boolean hasDoorBlocks()
+	{
+		return !blocks.isEmpty() && blocks.size() >= 2;
```
Comment (TheTurkeyDev) [inline Core/src/main/java/com/theprogrammingturkey/comz/game/features/Door.java:190]: Pretty sure you only need `blocks.size() >= 2` as that also infers the array isn't empty

File: Core/src/main/java/com/theprogrammingturkey/comz/game/actions/DoorSetupAction.java (hunk 1 above)
Comment (ikbenignace) [reply Core/src/main/java/com/theprogrammingturkey/comz/game/actions/DoorSetupAction.java:86]: It just need to be removed

File: Core/src/main/java/com/theprogrammingturkey/comz/game/features/Door.java (hunk 2 above)
Comment (ikbenignace) [reply Core/src/main/java/com/theprogrammingturkey/comz/game/features/Door.java]: Done

File: Core/src/main/java/com/theprogrammingturkey/comz/game/actions/DoorSetupAction.java (hunk 1 above)
Comment (ikbenignace) [reply Core/src/main/java/com/theprogrammingturkey/comz/game/actions/DoorSetupAction.java]: Done

**Pull request merged**